File: Java/Linked List/Singly Circular Linked List/SinglyCL03.java
/*
=====================================================================

Name            : Singly Linked List - Node Constructor 
Program Number  : 003
Description     : This program demonstrates the use of a
                  parameterized constructor in a node class.

                  The constructor initializes:
                  - Data field
                  - Next reference

Concepts Used   : Constructor, Object Initialization,
                  this Keyword, Node Creation

Input           : Integer value passed to constructor

Output          : Data and next reference value

Date            : July 2026
Language        : Java
Author          : Atharv Tushar Bhosale

=====================================================================
*/

class node
{
    public int data;
    public node next;

    node(int no)
    {
        this.data = no;
        this.next = null;
    }
}

class SinglyCL03
{
    public static void main(String A[])
    {
        node newn = new node(11);

        System.out.println(newn.data);
        System.out.println(newn.next);
    }
}

/*
=====================================================================

Sample Output :

11
null

=====================================================================

Observation :

- Constructor initializes node automatically.
- Data is assigned during object creation.
- Next reference is initialized to null.
- Reduces manual initialization code.

=====================================================================
*/

File: C++/Linked List/Singly Circular Linked List/SinglyCL03.cpp
/*
=====================================================================

Name            : Singly Circular Linked List - Manual Linking
Program Number  : 003

Description     : This program demonstrates linking multiple nodes
                  together using next pointers.

                  The list is linear and not circular yet.

Concepts Used   : Node Linking,
                  Pointer Assignment,
                  Linked Structure

Input           : None

Output          : Displays linked node data

Date            : July 2026
Language        : C++
Author          : Atharv Tushar Bhosale

=====================================================================
*/

#include<iostream>
using namespace std;

struct node
{
    int data;
    struct node* next;
};

typedef struct node NODE;
typedef struct node* PNODE;

int main()
{
    PNODE first = new NODE;
    PNODE second = new NODE;
    PNODE third = new NODE;

    first->data = 11;
    second->data = 21;
    third->data = 51;

    first->next = second;
    second->next = third;
    third->next = NULL;

    cout<<first->data<<endl;
    cout<<second->data<<endl;
    cout<<third->data<<endl;

    return 0;
}

/*
=====================================================================

Output :

11
21
51

=====================================================================

*/
